server/src/test/java/io/crate/integrationtests/TransportSQLActionTest.java

```diff
@@ -1581,6 +1581,15 @@ public void testWeirdIdentifiersAndLiterals() throws Exception {
             .hasRows("'");
     }
 
+    @Test
+    public void test_negated_and_three_value_query() throws Exception {
+        execute("CREATE TABLE t1(a INT, b INT)");
+        execute("INSERT INTO t1(a, b) VALUES (null, -1), (null, null), (-1, -1), (1, 1), (null, 1)");
+        execute("refresh table t1");
+        execute("SELECT * FROM t1 WHERE NOT (a AND b) order by a");
+        assertThat(response).hasRows("-1| -1", "NULL| -1");
+    }
+
```

Comment:
Can we write a cheaper test case in `NotPredicateTest` for this?
No need to start full nodes to test the query generation & matching

Reply:
ok